libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -402,26 +404,30 @@ bool AP_DDS_Client::start(void)
 }
 
 // read function triggered at every subscription callback
-void AP_DDS_Client::on_topic (uxrSession* uxr_session, uxrObjectId object_id, uint16_t request_id, uxrStreamId stream_id, struct ucdrBuffer* ub, uint16_t length, void* args)
+void AP_DDS_Client::on_topic_trampoline(uxrSession* uxr_session, uxrObjectId object_id, uint16_t request_id, uxrStreamId stream_id, struct ucdrBuffer* ub, uint16_t length,
+                                        void* args)
+{
+    AP_DDS_Client *dds = (AP_DDS_Client *)args;
+    dds->on_topic(uxr_session, object_id, request_id, stream_id, ub, length);
+}
+
+void AP_DDS_Client::on_topic(uxrSession* uxr_session, uxrObjectId object_id, uint16_t request_id, uxrStreamId stream_id, struct ucdrBuffer* ub, uint16_t length)
 {
-    (void) uxr_session; (void) object_id; (void) request_id; (void) stream_id; (void) length;
     /*
     TEMPLATE for reading to the subscribed topics
     1) Store the read contents into the ucdr buffer
     2) Deserialize the said contents into the topic instance
     */
-
+    (void) uxr_session; (void) request_id; (void) stream_id; (void) length;
     switch (object_id.id) {
-    case topics[to_underlying(TopicIndex::JOY_SUB)].dr_id.id:
+    case topics[to_underlying(TopicIndex::JOY_SUB)].dr_id.id: {
         sensor_msgs_msg_Joy topic;
         const bool success = sensor_msgs_msg_Joy_deserialize_topic(ub, &topic);
-
         if (success == false) {
             break;
         }
 
-        uint32_t* count_ptr = (uint32_t*) args;
-        (*count_ptr)++;
+        subscribe_sample_count++;
         if (topic.axes_size >= 4) {
             GCS_SEND_TEXT(MAV_SEVERITY_INFO,"Received sensor_msgs/Joy: %f, %f, %f, %f",
```

Comment:
these messages will be very annoying. You could have a DDS_OPTIONS parameter with a bit which enables debug. Otherwise add a debug() macro. See AP_BLHeli.cpp for a debug macro example, it would default off

Reply:
These were planned to be removed once the implementation of RC input is complete.